### PR TITLE
Fix CSV problems

### PR DESCRIPTION
**Pull request by breck7**
I just copy pasted the jtree library in for now to keep it simple and avoid any external dependencies. What do you think? Happy to approach it differently if you'd prefer.

### CODE REVIEW

File: sqlcaster.js [hunk 1]
```diff
@@ -127,16 +127,15 @@ document.addEventListener("DOMContentLoaded", function () {
...
-        "data:text/csv;charset=utf-8,\n" +
-        Object.keys(CURRENT_RESULTS[0]).join(",") +
-        "\n";
-      for (let row of CURRENT_RESULTS) {
-        csvContent += Object.values(row).join(",") + "\n";
-      }
-      const encodedUri = encodeURI(csvContent);
-      window.open(encodedUri);
+    downloadButton.addEventListener("click", (evt) => {
+      console.log(CURRENT_RESULTS);
```
Comment (breck7) [inline sqlcaster.js]: oops forgot to take this line out

File: lib/jtree.browser.js [hunk 2]
```diff
@@ -0,0 +1,6126 @@
+"use strict"
```
Comment (whatrocks) [inline lib/jtree.browser.js]: Can you add a comment here explaining that we've copy pasted this from this library and add a link to the library and this specific file in their source control?

File: sqlcaster.js [hunk 3]
```diff
@@ -127,16 +127,15 @@ document.addEventListener("DOMContentLoaded", function () {
...
+      window.open(
+        encodeURI(
+          `data:text/csv;charset=utf-8,\n${new TreeNode(
```
Comment (whatrocks) [inline sqlcaster.js:133]: What does `jtree` library do here that solves our problem? I'm not familiar with it and not sure if I've found the correct one on npm. I was thinking about pulling in a csv specific library, so curious about this approach.

**Review by whatrocks — COMMENTED** [review]
Left two questions/comments. 

Also, does this library offer CDN style source that we can just link out to in our html .. rather than the copy-pasta?

**Comment by whatrocks** [comment]
Oh, and does it solve the problem. Did you manually test it out?

Thanks!!

File: lib/jtree.browser.js (hunk 2 above)
Comment (breck7) [reply lib/jtree.browser.js]: Yes good idea. I can do it after pull

File: sqlcaster.js (hunk 3 above)
Comment (breck7) [reply sqlcaster.js:133]: jtree is a swiss army knife for data. The docs are pretty weak but it powers PLDB.com CancerDB.com Scroll.pub Ohayo.computer and many many more. Our company is putting a lot of resources into it from now on so should be rock solid.

**Comment by breck7** [comment]
> Also, does this library offer CDN style source that we can just link out to in our html .. rather than the copy-pasta?

Yes absolutely it does. I did copy-pasta for 2 reasons:

1. You have a very fast development experiment right now because it's just `git clone` with no build step. I'd recommend keeping that for another few days at least until your project starts to get big enough to require a build step.
2. I didn't want to add a dependency to an online repo. Although, come to think of it, you already depend on internet connection right? So perhaps we can just do an include.


What do you think about option 2? Maybe you are right and we should just do CDN right now.

If we went with CDN style the tag is this: `<script src="https://jtree.treenotation.org/products/jtree.browser.js"></script>`

File: sqlcaster.js (hunk 3 above)
Comment (whatrocks) [reply sqlcaster.js:133]: is it on npm right now?

**Comment by whatrocks** [comment]
> What do you think about option 2? Maybe you are right and we should just do CDN right now.
> 
> If we went with CDN style the tag is this: `<script src="https://jtree.treenotation.org/products/jtree.browser.js"></script>`

Yeah, let's do this approach then.

Did you happen to test it out? Does this fix the issue?

File: sqlcaster.js (hunk 3 above)
Comment (breck7) [reply sqlcaster.js:133]: yes https://www.npmjs.com/package/jtree

**Comment by breck7** [comment]
> Did you happen to test it out? Does this fix the issue?

Yes.

Before:
Screen Shot 2022-10-04 at 7 11 11 PM



After:
Screen Shot 2022-10-04 at 7 11 42 PM

**Comment by breck7** [comment]
@whatrocks I think this is ready to squash and merge in but I'll let you make the call 

Bonus: its -1 LOC :)

**Review by whatrocks — APPROVED** [review]
Looks great!

**Pull request merged**